### PR TITLE
【1.1.13】interface fileinfo add totalline

### PR DESCRIPTION
**Pull request by v-kkhuang**
### What is the purpose of the change

interface fileinfo add totalline

### Brief change log

- Define the core abstraction and interfaces of the EngineConn Factory;
- Define the core abstraction and interfaces of Executor Manager.

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java [hunk 1]
```diff
@@ -518,23 +518,35 @@ public Message fileInfo(
...
       if (null != fileInfo && fileInfo.length > 0) {
+        int rowNumber = (int) fileInfo[0].getSecond();
         message.data("path", path);
         message.data("colNumber", fileInfo[0].getFirst());
         message.data("rowNumber", fileInfo[0].getSecond());
+        if (rowNumber >= pageSize) {
+          fileSource2 = FileSource$.MODULE$.create(fsPath, fileSystem);
+          fileSource2.collect();
+          message.data("totalLine", fileSource2.getTotalLine());
+          IOUtils.closeQuietly(fileSource2);
```
Comment (casionone) [inline linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java:536]: do in finally block

Comment (casionone) [reply linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java:536]: optimize filesource2 
add tool function of getTotleLine at org.apache.linkis.storage.utils.FileSystemUtils

Comment (v-kkhuang) [reply linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java:536]: done

File: linkis-commons/linkis-storage/src/main/scala/org/apache/linkis/storage/utils/FileSystemUtils.scala [hunk 2]
```diff
@@ -113,4 +114,15 @@ object FileSystemUtils extends Logging {
...
   }
 
+  /**
+   * 返回文件总行数
+   * @param fileSource
+   * @return
+   *   TotalLine
+   */
+  def getTotalLine(fileSource: FileSource): Int = {
+    fileSource.collect()
```
Comment (casionone) [inline linkis-commons/linkis-storage/src/main/scala/org/apache/linkis/storage/utils/FileSystemUtils.scala:124]: optimiz to FileSystemUtils.getTotalLine(fsPath, fileSystem))

Comment (v-kkhuang) [reply linkis-commons/linkis-storage/src/main/scala/org/apache/linkis/storage/utils/FileSystemUtils.scala:124]: done

**Pull request merged**